llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp

```diff
@@ -6966,6 +6966,10 @@ static Function *emitTargetTaskProxyFunction(OpenMPIRBuilder &OMPBuilder,
     auto *ArgStructType =
         dyn_cast<StructType>(ArgStructAlloca->getAllocatedType());
```

Comment:
I see that it does the same thing that is already done in the function but still, why can't we use a `cast` instead of `dyn_cast` + assert since it is how LLVM coding guidelines suggest to do?

Reply:
Thank you @Fznamznon for reviews!. Fixed now.